marimo/_runtime/packages/package_manager.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2024 Marimo. All rights reserved.
+from __future__ import annotations
+
+import importlib.util
+import os
+import subprocess
+import sys
+import urllib.request
+
+from marimo._ast.cell import CellId_t
+from marimo._runtime.dataflow import DirectedGraph
+from marimo._runtime.packages.module_name_to_pypi_name import (
+    MODULE_NAME_TO_PYPI_NAME,
+)
+from marimo._utils.platform import is_pyodide
+
+
+class PackageManager:
```

Comment:
could we make this an abstract class, and then have `PipPackageManager`. may feel like a pre-optimization, but we know its coming down the the road and will force a cleaner design

Reply:
I started doing this, but it felt premature. My preference is to abstract and refactor when we need to

Reply:
I agree, we should avoid pre-optimizations, but this 1) makes it easier to test, and 2) welcomes contributions for other managers 

Also 3) don't we need this in this PR for micropip/wasm?

Reply:
Ok, sounds good